test/functions/aeminitiateupload.test.js

```diff
@@ -221,14 +224,116 @@ describe("AEMInitiateUpload", () => {
             {
                 const { value, done } = await generator.next();
                 assert.deepStrictEqual(value, new TransferAsset(source2, target2, {
-                    metadata: new AssetMetadata(undefined, "image/png", 1234),
+                    metadata: new AssetMetadata("target2.png", "image/png", 5678),
                     multipartTarget: new AssetMultipart([
                         "http://host/path/to/target2.png/block"
                     ], 2000, 20000, undefined, "http://host/path/to.completeUpload.json", "uploadToken2")
                 }));
                 assert.strictEqual(done, false);    
             }
     
+            // ensure there are no more files
+            {
+                const { value, done } = await generator.next();
+                assert.strictEqual(value, undefined);
+                assert.strictEqual(done, true);    
+            }
+        });
+        it("single asset, no mimetype", async () => {
+            const source = new Asset("file:///path/to/source.png");
```

Comment:
are the 3 slashes on purpose?

Reply:
it's required because there is no host in the file url: https://en.wikipedia.org/wiki/File_URI_scheme